### PR TITLE
[SYCLomatic] Support query CUDA syntax sugar of --query-api-mapping.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

File: clang/lib/DPCT/APIMapping/QueryAPIMapping.cpp [hunk 1]
```diff
@@ -27,34 +27,33 @@ void APIMapping::registerEntry(std::string Name, llvm::StringRef SourceCode) {
...
-  // 2. For each name got by step 1, put 4 kind of fuzzed name into the map
+  // 1. Change "Name" to lower case. (Querying will change "Key" to lower too)
+  // 2. Remove partial or all suffix '_'.
+  std::transform(Name.begin(), Name.end(), Name.begin(), ::tolower);
+  while (Name.back() == '_') {
+    Name.erase(Name.end() - 1);
+    EntryMap.try_emplace(Name, TargetIndex);
+  }
+  const auto EmplaceWithAndWithoutSuffix =
+      [TargetIndex](std::string Name, const std::string Suffix) {
```
Comment (yihwang-nv) [inline clang/lib/DPCT/APIMapping/QueryAPIMapping.cpp]:
```suggestion
      [TargetIndex](StringRef Name,StringRef Suffix) {
```

Comment (tangjj11) [reply clang/lib/DPCT/APIMapping/QueryAPIMapping.cpp]: only can change the second one to `StringRef`.

Comment (yihwang-nv) [reply clang/lib/DPCT/APIMapping/QueryAPIMapping.cpp]: Maybe it would be better to use llvm::Twine::concat to concat string.

**Review by zhimingwang36 — APPROVED** [review]

File: clang/lib/DPCT/APIMapping/QueryAPIMapping.cpp (hunk 1 above)
Comment (tangjj11) [reply clang/lib/DPCT/APIMapping/QueryAPIMapping.cpp]: fixed

**Review by yihwang-nv — APPROVED** [review]
LGTM

**Pull request merged**